File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/aws/s3/ObjectKey.java
package uk.nhs.nhsx.core.aws.s3;

import uk.nhs.nhsx.core.ValueType;

public class ObjectKey extends ValueType<ObjectKey> {

    private ObjectKey(String value) {
        super(value);
    }

    public ObjectKey append(String suffix) {
        return ObjectKey.of(value + suffix);
    }

    public static ObjectKey of(String name) {
        return new ObjectKey(name);
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/aws/s3/ObjectKey.kt
package uk.nhs.nhsx.core.aws.s3

import dev.forkhandles.values.NonBlankStringValueFactory
import dev.forkhandles.values.StringValue

class ObjectKey private constructor(value: String) : StringValue(value) {
    fun append(suffix: String) = of(value + suffix)

    companion object : NonBlankStringValueFactory<ObjectKey>(::ObjectKey)
}
